docs/schemaValidationMiddleware.md

```diff
@@ -0,0 +1,126 @@
+below for examples of how different rules can handle different types of request data.
+
+
+The transformed request data is validated against the schema. Any validation errors are then transformed using
+`options.rules.validationErrorsToResponse` before they are sent to the user.  The default format looks roughly like:
+
+     {
+       ok: false,
+       message: "The JSON you have provided is not valid.",
+       errors: {
+         field1: ["This field is required."]
+       }
+     }
+
+The rejection output of this middleware is delivered using a [`schemaHandler`](./handler.md). It should match the JSON
+Schema specified in `options.responseSchemaKey` and `options.responseSchemaUrl`.
+
+### Component Options
+
+The following component configuration options are supported:
+
+| Option              | Type     | Description |
+| ------------------- | -------- | ----------- |
+| `handlerGrades` | `Array` | An array of grade names that will be used in constructing our request handler. |
+| `method` | `String` | The method(s) the inner router will respond to.  These should be lowercase strings corresponding to the methods exposed by Express routers.  The default is to use the `POST` method, there are convenience grades for each method. |
+| `responseSchemaKey` | `String` | The schema key that [our handler](./handler.md) will use in constructing response headers. |
+| `responseSchemaUrl` | `String` | The base URL where `responseSchemaKey` can be found. |
+| `rules.requestContentToValidate` | `Object` | The [rules to use in transforming](http://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html#fluid-model-transformwithrules-source-rules-options-)
+the incoming data before validation (see below for more details). |
+| `rules.validationErrorsToResponse` | `Object` | The [rules to use in transforming](http://docs.fluidproject.org/infusion/development/ModelTransformationAPI.html#fluid-model-transformwithrules-source-rules-options-)
+validation errors before they are sent to the user (see above). |
+| `schemaKey`         | `String` |  The key (also the filename) of the schema to be used for validation. |
+| `schemaDirs`        | `String` | The path to the schema directories that contain a file matching `options.schemaKey`.  This is expected to be an array of package-relative paths such as `%gpii-handlebars/tests/schemas`. |
+
+The default `rules.requestContentToValidate` in this grade are intended for use with `PUT` or `POST` body data.  This
+can be represented as follows:
+
+      requestContentToValidate: {
+          "": "body"
+      }
+
+See `gpii.schema.validationMiddleware.handlesGetMethod` below for an example of working with query data.
+
+### Invokers
+
+#### `{middleware}.middleware(req, res, next)`
+
+* `req`: The [request object](http://expressjs.com/en/api.html#req) provided by Express, which wraps node's [`http.incomingMessage`](https://nodejs.org/api/http.html#http_class_http_incomingmessage).
```

Comment:
See comments on other projects - this is not an example of wrapping - the request object *is* the incomingMessage instance